### PR TITLE
impl(bigquery): set all default options in a single place

### PR DESCRIPTION
**Pull request by jsrinnn**
Changes to address github issue: https://github.com/googleapis/google-cloud-cpp/issues/11026

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11038)

### CODE REVIEW

File: google/cloud/bigquery/v2/minimal/internal/job_request.cc [hunk 1]
```diff
@@ -35,11 +35,13 @@ StatusOr<rest_internal::RestRequest> BuildRestRequest(GetJobRequest const& r,
     return internal::InvalidArgumentError(
         "Invalid GetJobRequest: Job Id is empty", GCP_ERROR_INFO());
   }
-  // Fix the endpoints prefix and suffixes.
-  std::string endpoint = "https://bigquery.googleapis.com/";
-  if (opts.has<EndpointOption>()) {
-    endpoint = opts.get<EndpointOption>();
+  if (!opts.has<EndpointOption>()) {
```
Comment (coryan) [inline google/cloud/bigquery/v2/minimal/internal/job_request.cc:38]: nit: I am not convinced this is necessary, but cannot say that it hurts anything.

**Review by coryan — APPROVED** [review]

**Comment by coryan** [comment]
You may want to read about GitHub's keywords:

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Review by jsrinnn — COMMENTED** [review]
Ah...nice. Thanks!

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11038#-:-NQ__E0-18YkslnmVN3i:bgry0hz)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @coryan)

___
*[`google/cloud/bigquery/v2/minimal/internal/job_request.cc` line 38 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11038#-NQWRdMB-eGB_msHsE0A-r1-38:-NQ_Yxn62YYPdvsMbuXm:b-crxhri) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7d5adad2f927c360624058441aa1420591726fda/google/cloud/bigquery/v2/minimal/internal/job_request.cc#L38)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

nit: I am not convinced this is necessary, but cannot say that it hurts anything.
</blockquote></details>

I have addeed input validation checks for the presence of all required parameters where we know the api will fail without them.

**Review by devbww — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11038)* status: 0 of 4 files reviewed, 1 unresolved discussion

___
*[`google/cloud/bigquery/v2/minimal/internal/job_request.cc` line 38 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11038#-NQWRdMB-eGB_msHsE0A-r1-38:-NQaCuPlDNcV2wBTLNki:b8dxg5b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6e2927d32037bd249441d8d999b66679b29e775d/google/cloud/bigquery/v2/minimal/internal/job_request.cc#L38)):*
<details><summary><i>Previously, jsrinnn wrote…</i></summary><blockquote>

I have addeed input validation checks for the presence of all required parameters where we know the api will fail without them.
</blockquote></details>

For what it's worth, this is the opposite of what I would suggest.  That is, I would remove all input validation checks where we know the API will fail without them.  It duplicates tests, possibly inventing new error values, and is fragile in the case of changing requirements in the backend.

**Pull request merged**